src/FormElement/ScheduleElement/Common/FieldsUtils.php

```diff
@@ -0,0 +1,241 @@
+            'TH' => $this->translate('Thursday'),
+            'FR' => $this->translate('Friday'),
+            'SA' => $this->translate('Saturday'),
+            'SU' => $this->translate('Sunday')
+        ];
+    }
+
+    protected function createOrdinalElement(): FormElement
+    {
+        return $this->createElement('select', 'ordinal', [
+            'class'   => 'autosubmit',
+            'value'   => $this->getPopulatedValue('ordinal', static::$first),
+            'options' => [
+                static::$first  => $this->translate('First'),
+                static::$second => $this->translate('Second'),
+                static::$third  => $this->translate('Third'),
+                static::$fourth => $this->translate('Fourth'),
+                static::$fifth  => $this->translate('Fifth'),
+                static::$last   => $this->translate('Last')
+            ]
+        ]);
+    }
+
+    protected function createOrdinalSelectableDays(): FormElement
+    {
+        return $this->createElement('select', 'day', [
+            'class'   => 'autosubmit',
+            'value'   => $this->getPopulatedValue('day', static::$everyDay),
+            'options' => [
+                'Regular'       => $this->regulars,
+                'Non Standards' => [
+                    static::$everyDay     => $this->translate('Day'),
+                    static::$everyWeekday => $this->translate('Weekday (Mon - Fri)'),
+                    static::$everyWeekend => $this->translate('WeekEnd (Sat or Sun)')
+                ]
+            ]
+        ]);
+    }
+
+    /**
+     * Load the given RRule instance into a list of key=>value pairs
+     *
+     * @param RRule $rule
+     *
+     * @return array
+     */
+    public function loadRRule(RRule $rule): array
+    {
+        $values = [];
+        $isMonthly = $rule->getFrequency() === RRule::MONTHLY;
+        if ($isMonthly && (! empty($rule->getByMonthDay()) || empty($rule->getByDay()))) {
+            foreach ($rule->getByMonthDay() ?? [] as $value) {
+                $values["day$value"] = 'y';
+            }
+
+            $values['runsOn'] = MonthlyFields::RUNS_EACH;
+        } else {
+            $position = $rule->getBySetPosition();
+            $byDay = $rule->getByDay() ?? [];
+
+            if ($isMonthly) {
+                $values['runsOn'] = MonthlyFields::RUNS_ONTHE;
+            } else {
+                $months = $rule->getByMonth();
+                if (empty($months) && $rule->getStartDate()) {
+                    $months[] = $rule->getStartDate()->format('m');
+                } elseif (empty($months)) {
+                    $months[] = date('m');
+                }
+
+                $values['month'] = strtoupper($this->getMonthByNumber((int)$months[0]));
+                $values['runsOnThe'] = ! empty($byDay) ? 'y' : 'n';
+            }
+
+            if (count($byDay) == 1 && preg_match('/^(-?\d)+(\S.*)$/', $byDay[0], $matches)) {
```

Comment:
`(-?\d+)`?

It's slightly more efficient.

Reply:
But it doesn't match what I would like to do. Test string: `-1MO` or `2SA`

Reply:
It replaces only `(-?\d)+` of course :wink:

Reply:
Even then, my regex still wouldn't be the same. If you're only concerned about the performance, it's only `0.1ms` faster than my version.